### PR TITLE
New join move candidate

### PR DESCRIPTION
**Pull request by Polsonby**
## Description
This includes all the changes from 
https://github.com/ministryofjustice/correspondence_tool_staff/pull/1050
https://github.com/ministryofjustice/correspondence_tool_staff/pull/1067
and fixes the minor issues that were breaking the tests. 

This PR gives us the working join BU journey with front end styling as well as the revised BU journey

## Self-review checklist

* [x] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [x] (3) Tests passing
* [x] (4) Branch ready to be merged (not work in progress)
* [x] (5) No superfluous changes in diff
* [x] (6) No TODO's without new ticket numbers

### CODE REVIEW

**Comment by Polsonby** [comment]
Because these commits have already been worked on by Javid, signed off by Alex, and reviewed by Dan, I'm going to go ahead and use "god mode" to merge to master so I can work on a stable base

File: config/locales/en.yml [hunk 1]
```diff
@@ -1018,6 +1018,7 @@ en:
       subhead: "Business unit: %{team_name}"
       team_your_working_with: This is the team you're working with
       will_be_joined: "%{team_name}'s users and open cases will be joined to %{target_team_name}'s business unit"
+      cannot_join_other_team: "%{team_name} cannot be joined with any other business unit"
```
Comment (digitalali-moj) [inline config/locales/en.yml:1021]: Are we happy to not keep this alphabetical, personally not an issue

Comment (Polsonby) [reply config/locales/en.yml:1021]: Oh good spot - there might be a few of those. If you can mark with comments like this I'll tidy them up

File: config/locales/en.yml [hunk 2]
```diff
@@ -997,12 +1001,39 @@ en:
...
+        paragraph: "Moving a business unit means:"
       moved_successfully: "%{team_name} has been moved to %{destination_directorate_name}"
-      confirm_message: Are you sure you want to do this? This business unit will be deactivated to preserve its history, and a new one will be created
-
+      moving_to: Moving to
+      subhead: "Business unit: %{team_name}"
+      team_currently_located: This is where the team is currently located
+      will_be_moved: "%{team_name}'s users and open cases will be moved to %{target_directorate_name}"
+    unreversible: This action cannot be reversed
+    team: Business unit
```
Comment (digitalali-moj) [inline config/locales/en.yml:1036]: Just this other one for which I think I done myself should probably be one line up, can't notice any others

**Review by njpreddy — APPROVED** [review]
LGTM

**Pull request merged**